File: hello.zig
const std = @import("std");
const c = @cImport({
    // See https://github.com/zig-lang/zig/issues/515
    @cDefine("_NO_CRT_STDIO_INLINE", "1");
    @cInclude("stdio.h");
    @cInclude("newplus/plus.h");
});

pub fn main(init: std.process.Init) !void {
    const args = try init.minimal.args.toSlice(init.arena.allocator());

    if (args.len == 1) {
        _ = c.printf("First arg (0 - 2000000000) is required.\n");
        return;
    }

    const count = try std.fmt.parseInt(i32, args[1], 10);
    if (count <= 0 or count > 2000000000) {
        _ = c.printf("Must be a positive number not exceeding 2 billion.\n");
        return;
    }

    run(count);
}

fn run(count: i32) void {
    const start = c.current_timestamp();

    var x: i32 = 0;
    while (x < count) {
        x = c.plusone(x);
    }

    const elapsed = c.current_timestamp() - start;
    _ = c.printf("%lld\n", elapsed);
}

File: hello.dart
import 'dart:ffi' as ffi;
import 'dart:io' show File;

typedef _PlusOneNative = ffi.Int Function(ffi.Int);
typedef _PlusOne = int Function(int);
typedef _CurrentTimestampNative = ffi.LongLong Function();
typedef _CurrentTimestamp = int Function();

final _newplus = ffi.DynamicLibrary.open(
  File('newplus/libnewplus.so').absolute.path,
);
final _plusone = _newplus.lookupFunction<_PlusOneNative, _PlusOne>('plusone');
final _currentTimestamp = _newplus
    .lookupFunction<_CurrentTimestampNative, _CurrentTimestamp>(
      'current_timestamp',
    );

void run(int count) {
  // start
  var start = _currentTimestamp();

  int x = 0;
  while (x < count) {
    x = _plusone(x);
  }

  print(_currentTimestamp() - start);
}

void main(List<String> args) {
  if (args.isEmpty) {
    print("First arg (0 - 2000000000) is required.");
    return;
  }

  var count = int.parse(args[0]);
  if (count <= 0 || count > 2000000000) {
    print("Must be a positive number not exceeding 2 billion.");
    return;
  }

  _plusone(_currentTimestamp() == 0 ? 1 : 2);
  run(count);
}
